### PR TITLE
AnneJu step 10: Sort the task list in order of creation date and time

### PR DESCRIPTION
**Pull request by anneju**
#### Description
- As title

#### Story
- [x] Create sort form
- [x] Send sort params within Tasks#index page

#### Related RSpecs
```
rspec spec/system/tasks_spec.rb:31
```

### CODE REVIEW

File: myapp/app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+    'id' => DEFAULT_SORT,
+    'created_at_desc' => 'created_at DESC'
+  }
+
   before_action :set_task, only: %i[show edit update destroy]
 
   def index
-    @tasks = Task.all
+    order = ORDER_MAPPING[params[:order_by]] || DEFAULT_SORT
+    @tasks = Task.all.reorder(order)
```
Comment (stakahashi-d) [inline myapp/app/controllers/tasks_controller.rb]: Running query is same, so you can simply use order for here!

```suggestion
    @tasks = Task.all.order(order)
```

File: myapp/config/environments/development.rb [hunk 2]
```diff
@@ -59,4 +59,7 @@
   # Use an evented file watcher to asynchronously detect changes in source code,
   # routes, locales, etc. This feature depends on the listen gem.
   config.file_watcher = ActiveSupport::EventedFileUpdateChecker
+
+  # turn on web sonsole for debug
+  config.web_console.permissions = '172.20.0.1'
```
Comment (stakahashi-d) [inline myapp/config/environments/development.rb:64]: nice 😄

File: myapp/app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -1,10 +1,17 @@
 # frozen_string_literal: true
 
 class TasksController < ::ApplicationController
+  DEFAULT_SORT = 'id ASC'
+  ORDER_MAPPING = {
```
Comment (stakahashi-d) [inline myapp/app/controllers/tasks_controller.rb]: We have to add both `ASC` and `DESC` for keys in this case to add sort column in future.
So, may be separating with `ORDER_FIELD ` and `SORT_TYPE` is better. ←It's just a sample

``` .rb
ORDER_FIELD = ['id', 'created_at']
SORT_TYPE = ['asc', 'desc']
```

File: myapp/spec/system/tasks_spec.rb [hunk 4]
```diff
@@ -27,6 +27,32 @@
           expect(page).to have_content task_2.description
         end
       end
+
+      context 'when sort tasks' do
+        let!(:task_1) { create(:task, title: 'Task 1') }
+        let!(:task_2) { create(:task, title: 'Task 2', created_at: DateTime.now + 1) }
+        let(:sorted_tasks_ids) { page.all('.task-id').map(&:text).map(&:to_i) }
+
+        context 'by ID' do
```
Comment (stakahashi-d) [inline myapp/spec/system/tasks_spec.rb:44]: maybe you can add the test case for default order!

**Review by stakahashi-d — COMMENTED** [review]
Quick at work!
I added some comments, so please have a look!

File: myapp/app/controllers/tasks_controller.rb (hunk 1 above)
Comment (anneju) [reply myapp/app/controllers/tasks_controller.rb]: 👌🏻

Comment (anneju) [reply myapp/app/controllers/tasks_controller.rb:14]: 👌🏻

File: myapp/spec/system/tasks_spec.rb (hunk 4 above)
Comment (anneju) [reply myapp/spec/system/tasks_spec.rb:44]: sure, that's better~

File: myapp/app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -1,10 +1,13 @@
 # frozen_string_literal: true
 
 class TasksController < ::ApplicationController
+  SORTABLE_FIELDS = %w(id created_at)
+  SORTABLE_ORDERS = %w(asc desc)
```
Comment (anneju) [inline myapp/app/controllers/tasks_controller.rb:5]: not sure I get what you mean [here](https://github.com/Fablic/training/pull/1470#discussion_r1051886053), is that something like this...?

Comment (stakahashi-d) [reply myapp/app/controllers/tasks_controller.rb:5]: Yeah, something like that! 
Thanks for fixing

**Review by stakahashi-d — APPROVED** [review]
LGTM

**Pull request merged**